test/Path/Gen.hs

```diff
@@ -45,7 +45,6 @@ instance Validity (Path Rel Dir) where
     && FilePath.hasTrailingPathSeparator fp
     && FilePath.isValid fp
     && not (null fp)
-    && fp /= "."
     && fp /= ".."
     && not (".." `isInfixOf` fp)
     && (parseRelDir fp == Just p)
```

Comment:
This line need to be changed too now. Maybe it should even be refactored into a idempotency property:

```
parse (toFilePath p) == Just p
```